src/System.Management.Automation/engine/lang/parserutils.cs

```diff
@@ -978,66 +979,101 @@ internal static object ReplaceOperator(ExecutionContext context, IScriptExtent e
-        /// <param name="input">The input string.</param>
-        /// <param name="regex">A Regex instance.</param>
-        /// <param name="substitute">The substitute value.</param>
-        /// <returns>The result of the regex.Replace operation.</returns>
-        private static object ReplaceOperatorImpl(ExecutionContext context, string input, Regex regex, object substitute)
+        private struct OperatorReplacer
         {
-            switch (substitute)
+            public static OperatorReplacer Create(ExecutionContext context, Regex regex, object substitute)
             {
-                case string replacementString:
-                    return regex.Replace(input, replacementString);
+                return new OperatorReplacer(context, regex, substitute, replacementString: null, matchEvaluator: null);
+            }
+
+            private readonly ExecutionContext _context;
+            private readonly Regex _regex;
+
+            private string _cachedReplacementString;
+
+            private MatchEvaluator _cachedMatchEvaluator;
+
+            private OperatorReplacer(
+                ExecutionContext context,
+                Regex regex,
+                object substitute,
+                string replacementString,
+                MatchEvaluator matchEvaluator)
+            {
+                _context = context;
+                _regex = regex;
+                _cachedReplacementString = replacementString;
+                _cachedMatchEvaluator = matchEvaluator;
+
+                switch (substitute)
+                {
+                    case string replacement:
+                        _cachedReplacementString = replacement;
+                        break;
+
+                    case ScriptBlock sb:
+                        _cachedMatchEvaluator = GetMatchEvaluator(_context, sb);
+                        break;
 
-                case ScriptBlock sb:
-                    MatchEvaluator me = GetMatchEvaluator(context, sb);
-                    return regex.Replace(input, me);
+                    case object val when LanguagePrimitives.TryConvertTo(val, out _cachedMatchEvaluator):
+                        break;
 
-                case object val when LanguagePrimitives.TryConvertTo(val, out MatchEvaluator matchEvaluator):
-                    return regex.Replace(input, matchEvaluator);
+                    default:
+                        _cachedReplacementString = PSObject.ToStringParser(_context, substitute);
+                        break;
+                }
 
-                default:
-                    string replacement = PSObject.ToStringParser(context, substitute);
-                    return regex.Replace(input, replacement);
+                // Local helper function to avoid creating an instance of the generated delegate helper class
+                // every time 'ReplaceOperatorImpl' is invoked.
+                static MatchEvaluator GetMatchEvaluator(ExecutionContext context, ScriptBlock sb)
```

Comment:
Now, it's better make this method directly a static method of the struct.

Reply:
Done.